devicetokens/devicetokens.go

```diff
@@ -0,0 +1,84 @@
+package devicetokens
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/tidepool-org/platform/structure"
+)
+
+type Document struct {
+	// UserID of the user that owns the DeviceToken.
+	UserID string `json:"userId" bson:"userId"`
+	// TokenID is string that uniquely identifies the DeviceToken.
+	//
+	// It's useful for generating unique indexes.
+	TokenID string `json:"tokenId" bson:"tokenId"`
+	// DeviceToken wraps the device-specific token.
+	DeviceToken DeviceToken `json:"deviceToken" bson:"deviceToken"`
+}
+
+func NewDocument(userID string, deviceToken DeviceToken) *Document {
```

Comment:
We really need some validation that we aren't introducing bogus data into the database. We can do it here or in the API via the `Validate` function.

Reply:
Discussed below: https://github.com/tidepool-org/platform/pull/667#discussion_r1425875589